ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -745,23 +835,22 @@ mempoolCapTestExpectedTrace mcts = go chunks
     txs :: [GenTx TestBlock]
     txs = map TestGenTx mctsValidTxs
 
-    chunksOf :: Int -> [a] -> [[a]]
-    chunksOf _ [] = []
-    chunksOf n xs
-      | n > 0     = take n xs : chunksOf n (drop n xs)
-      | otherwise = error $  "mempoolCapTestExpectedTrace.chunksOf: "
-                          <> "n is less than or equal to 0"
+    buildChunks :: [[GenTx TestBlock]] -> [GenTx TestBlock] -> [[GenTx TestBlock]]
+    buildChunks acc []     = acc
+    buildChunks acc remTxs =
+      let (ts, ts') = splitTxsUntilCap remTxs (MempoolCapacity mempoolCap)
```

Comment:
No need to unwrap the `MempoolCapacity` constructor on L832 and wrap it here again.